CRM/Core/BAO/Dashboard.php

```diff
@@ -399,33 +399,20 @@ public static function addDashlet(&$params) {
     $dashlet = new CRM_Core_DAO_Dashboard();
 
     if (!$dashboardID) {
-
       // Assign domain before search to allow identical dashlets in different domains.
-      if (empty($params['domain_id'])) {
-        $dashlet->domain_id = CRM_Core_Config::domainID();
-      }
-      else {
-        $dashlet->domain_id = $params['domain_id'] ?? NULL;
-      }
+      $dashlet->domain_id = $params['domain_id'] ?? CRM_Core_Config::domainID();
 
       // Try and find an existing dashlet - it will be updated if found.
-      if (!empty($params['name'])) {
+      if (!empty($params['name']) || !empty($params['url'])) {
         $dashlet->name = $params['name'] ?? NULL;
-        $dashlet->find(TRUE);
-      }
-      else {
         $dashlet->url = $params['url'] ?? NULL;
         $dashlet->find(TRUE);
       }
-
     }
     else {
       $dashlet->id = $dashboardID;
     }
 
-    if (isset($params['permission']) && is_array($params['permission'])) {
```

Comment:
I am assuming you removed this to rely on the copyValues but the field doesn't have a serialize type in the schema so that won't happen

Reply:
ah sorry  - you DID add it - I thought the above change was that DAO